core/indigo-core/common/math/algebra.h

```diff
@@ -303,6 +303,54 @@ namespace indigo
             _rightTop = other._rightTop;
         }
 
+        inline bool pt_in_rect(const Vec2f& pt) const
+        {
+            return _leftBottom.x < pt.x && _leftBottom.y < pt.y && _rightTop.x > pt.x && _rightTop.y > pt.y;
+        }
+
+        inline bool ray_intersects_rect(const Vec2f& begin, const Vec2f& end)
+        {
+            Vec2f v = end - begin;
+            Vec2f vr(v.y, -v.x); // perpendicular vector
+            auto lb = _leftBottom - begin;
+            auto lt = leftTop() - begin;
+            auto rt = _rightTop - begin;
+            auto rb = rightBottom() - begin;
+            // same_sign means no intersection
+            bool same_sign = std::signbit(Vec2f::dot(vr, lb)) == std::signbit(Vec2f::dot(vr, lt)) &&
+                             std::signbit(Vec2f::dot(vr, lt)) == std::signbit(Vec2f::dot(vr, rt)) &&
+                             std::signbit(Vec2f::dot(vr, rt)) == std::signbit(Vec2f::dot(vr, rb));
+
+            return !same_sign && Vec2f::cross(lb, vr) > 0 && Vec2f::cross(lt, vr) > 0 && Vec2f::cross(rt, vr) > 0 && Vec2f::cross(rb, vr) > 0;
+        }
+
+#define HYPOT(a, b) (sqrt((a) * (a) + (b) * (b)))
+
+        inline double pt_distance(const Vec2f& pt)
```

Comment:
Looking at other names, here was used camelCase style, not Snake case. Let's follow the style of the header.

Reply:
fixed